wasmtime-environ/src/cache/worker.rs

```diff
@@ -833,9 +873,8 @@ fn is_fs_lock_expired(
     allowed_future_drift: Duration,
 ) -> bool {
     let mtime = match entry
-        .map(|e| e.metadata())
-        .unwrap_or_else(|| path.metadata())
-        .and_then(|metadata| metadata.modified())
+        .map_or_else(|| path.metadata(), |e| e.metadata())
+        .and_then(|metadata| metadata.modified()) // todo: research if we can change it to creation time (for the tests), if yes - it would be a little better to use it
```

Comment:
As we talked about offline, we can remove this todo and just use the `modified()` time. If there is somehow a process continually appending to the cache, it'll eventually consume all of disk space and get noticed :-).

Reply:
Done.